cpp/daal/src/algorithms/dtrees/forest/classification/df_classification_train_dense_default_impl.i

```diff
@@ -905,6 +910,667 @@ void UnorderedRespHelper<algorithmFPType, cpu>::finalizeBestSplit(const IndexTyp
     if (bestSplit.featureValue == this->getValue(iFeature, iNext)) bestSplit.featureValue = this->getValue(iFeature, iRowSplitVal);
 }
 
+//////////////////////////////////////////////////////////////////////////////////////////
+// UnorderedRespHelperRandom class for random splitting classification
+//////////////////////////////////////////////////////////////////////////////////////////
+template <typename algorithmFPType, CpuType cpu>
+class UnorderedRespHelperRandom : public UnorderedRespHelper<algorithmFPType, cpu>
+{
+public:
+    typedef double intermSummFPType;
+    using Histogramm   = typename UnorderedRespHelper<algorithmFPType, cpu>::Histogramm;
+    using ImpurityData = typename UnorderedRespHelper<algorithmFPType, cpu>::ImpurityData;
+    using TSplitData   = typename UnorderedRespHelper<algorithmFPType, cpu>::TSplitData;
+
+public:
+    UnorderedRespHelperRandom(const dtrees::internal::IndexedFeatures * indexedFeatures, size_t nClasses)
+        : UnorderedRespHelper<algorithmFPType, cpu>(indexedFeatures, nClasses)
+    {}
+
+    bool findBestSplitForFeature(const algorithmFPType * featureVal, const IndexType * aIdx, size_t n, size_t nMinSplitPart,
```

Comment:
Please rename `findBestSplit` into `findSplit` in UnorderedRespHelper and UnorderedRespHelperRandom.

Reply:
New commits do not address this, as random and best algorithms chose best from set of evaluated features.